HW_3/task_2.py

```diff
@@ -0,0 +1,28 @@
+# Задача 18: Требуется найти в массиве A самый близкий по величине элемент к заданному числу X.
+# Если таких элементов несколько, вывести один любой. 
+# Пользователь в первой строке вводит натуральное число N – количество элементов в массиве.
+# В последующих  строках записаны N целых чисел Ai (можно использовать псевдогенерацию). Последняя строка содержит число X
+# *Пример:*
+# 5
+# 7 -2 3 5 1
+# 6
+# -> 7
+
+import random
+
+n = int(input('Введите количество элементов массива: '))
+
+a = [random.randint(0, 9) for i in range(n)]
+print(a, end = " ")
+
+x = int(input(f'\nЗадайте число X: '))
+
+с = e = 10
+
+for i in range(0, len(a)):
```

Comment:
Ноль передавать не обязательно. Во-вторых у вас уже существуют параметр, который содержит в длину списка.
```suggestion
for i in range(n):
```